### PR TITLE
install cni for x install

### PR DESCRIPTION
**Pull request by ZhengzheYang**

### CODE REVIEW

File: asmcli/commands/experimental/install.sh [hunk 1]
```diff
@@ -15,7 +15,8 @@ x_install() {
   local CLUSTER_LOCATION; CLUSTER_LOCATION="$(context_get-option "CLUSTER_LOCATION")"
   local CLUSTER_NAME; CLUSTER_NAME="$(context_get-option "CLUSTER_NAME")"
 
-  # TODO: MANAGED_CNI install should go here, if necessary
+  install_mananged_cni
```
Comment (bianpengyuan) [inline asmcli/commands/experimental/install.sh:18]: Let's add another TODO here to remove this when AFC based install is ready.

**Review by bianpengyuan — APPROVED** [review]

File: asmcli/commands/experimental/install.sh (hunk 1 above)
Comment (ZhengzheYang) [reply asmcli/commands/experimental/install.sh:19]: Done

File: asmcli/tests/common.sh [hunk 2]
```diff
@@ -854,3 +860,17 @@ EOF
   if [[ -n "${NOTFOUND}" ]]; then NOTFOUND="$(strip_trailing_commas "${NOTFOUND}")"; fi
   echo "${NOTFOUND}"
 }
+
+check_cni_ready() {
+  if ! kubectl get daemonset istio-cni-node -n kube-system 2>/dev/null; then
+    false
```
Comment (zerobfd) [inline asmcli/tests/common.sh:866]: false; return

**Review by zerobfd — APPROVED** [review]
LGTM if @bianpengyuan agrees that the test logic looks correct. One issue about `false` without returning.

File: asmcli/tests/common.sh [hunk 3]
```diff
@@ -763,6 +763,12 @@ run_basic_test() {
   label_with_revision "${LT_NAMESPACE}" "${LABEL}"
   roll "${LT_NAMESPACE}"
 
+  if [[ "${EXTRA_FLAGS}" = *--managed* || "${SUBCOMMAND}" = *experimental* ]]; then
+    for _ in {1..5}; do
```
Comment (bianpengyuan) [inline asmcli/tests/common.sh:767]: Not familiar with the code here. After five retry, how does the test fail?

Comment (ZhengzheYang) [reply asmcli/tests/common.sh:767]: This is WIP. Will update once I am done. Does the 5 retries and `sleep 10` make sense?

Comment (bianpengyuan) [reply asmcli/tests/common.sh:767]: Yep

**Review by bianpengyuan — APPROVED** [review]

**Pull request merged**